src/components/lec6/lec6task6/apierror.js

```diff
@@ -0,0 +1,50 @@
+
+import React, { Component } from "react";
+
+
+export default class ApiError extends Component {
+    constructor(props){
+        super(props);
+
+        this.state = {
+            posts: [],
+            isLoading: false,
+            error: null,
+          };
+    }
+
+    componentDidMount() {
+        this.setState({ isLoading: true });
```

Comment:
Instead of calling `this.setState` you can just make the default `state` true;

```javascript
this.state = {
  isLoading: true,// default value is true and when you get the response update the state.
}
```